rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -139,27 +142,30 @@ func Qualify_AssetClass_Equity:
     output:
         is_product boolean (1..1)
 
-    alias optionUnderlier: economicTerms -> payout -> optionPayout only-element -> underlier
+    alias optionUnderlier: economicTerms -> payout -> OptionPayout only-element -> underlier
     alias settlementUnderlier:
-        economicTerms -> payout -> settlementPayout only-element -> underlier
+        economicTerms -> payout -> SettlementPayout only-element -> underlier
 
     set is_product:
-        (economicTerms -> payout -> performancePayout -> underlier -> Observable
+        (economicTerms -> payout -> PerformancePayout -> underlier -> Observable
             extract Qualify_UnderlierObservable_Equity
             then all = True
                     and ( // Interest Rate Payout + Performance Payout (Price Return Swap, Total Return Swap)
-                    (economicTerms -> payout -> interestRatePayout, economicTerms -> payout -> performancePayout) only exists
+                        (economicTerms -> payout count = 2 
+                            and economicTerms -> payout -> InterestRatePayout exists 
+                            and economicTerms -> payout -> PerformancePayout exists)
                         // Performance Payout + Fixed Price Payout (Dividend Swap)
-                        or (economicTerms -> payout -> performancePayout, economicTerms -> payout -> fixedPricePayout) only exists
-                            // Performance Payout only (Variance, Volatility and Correlation Swap)
-                or economicTerms -> payout -> performancePayout only exists))
-            or (economicTerms -> payout -> optionPayout only exists
+                        or (economicTerms -> payout -> PerformancePayout exists 
+                            and economicTerms -> payout -> FixedPricePayout exists)
```

Comment:
This logic doesn't match current behaviour if it doesn't also assert that there are only 2 payouts.

Why not keep the current `only exists` syntax?

Reply:
I think dividend swaps can have 2 FixedPricePayouts?  So assert that there are 3 payouts in total?  Or, instead, assert that there is 1 PerformancePayout?

> Why not keep the current only exists syntax?
The current `only exists` syntax it not suitable for payouts anymore as it is a list, and each choice attribute has a different parent.

Reply:
To rewrite the previous logic, it would be:
```
economicTerms -> payout -> PerformancePayout exists 
and economicTerms -> payout -> FixedPricePayout exists
// check nothing else exists
and economicTerms -> payout [ PerformancePayout exists or FixedPricePayout exists ] all = True

```

Reply:
Updated as per discussions